ros2_control_demo_hardware/CMakeLists.txt

```diff
@@ -7,7 +7,7 @@ if(NOT CMAKE_CXX_STANDARD)
 endif()
 
 if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
-  add_compile_options(-Wall -Wextra)
+  add_compile_options(-Wall -Wextra -Werror)
```

Comment:
I think if we turn all warnings into errors, we should compile also with the `pedantic' flag.

Reply:
Ok, using recommendations https://docs.ros.org/en/foxy/Guides/Ament-CMake-Documentation.html#compiler-and-linker-options